From 798f07e9c716329339fe7b51ab7a82cf3b83ebcb Mon Sep 17 00:00:00 2001
From: Jan Nieuwenhuizen <janneke@gnu.org>
Date: Thu, 30 Jul 2009 15:53:47 +0200
Subject: [PATCH] OOXML: calc export: init SotStorage from shell.GetMedium.  Fixes NULL init.

While moving xlsx export to uno filter, SotStorage init code had to
be added.  While it compiled, it always produced a NULL pointer.
This inhibits any export using rStorage, such as pivotCache.

   * Modified     sc/source/filter/excel/xestream.cxx
---
 sc/source/filter/excel/xestream.cxx |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git sc/source/filter/excel/xestream.cxx sc/source/filter/excel/xestream.cxx
index 0568512..84d038c 100644
--- sc/source/filter/excel/xestream.cxx
+++ sc/source/filter/excel/xestream.cxx
@@ -58,6 +58,10 @@
 #include <formula/grammar.hxx>
 #include <oox/export/drawingml.hxx>
 
+#include <sfx2/docfile.hxx>
+#include <sfx2/objsh.hxx>
+#include <sfx2/app.hxx>
+
 #define DEBUG_XL_ENCRYPTION 0
 
 using ::std::vector;
@@ -1116,7 +1120,9 @@ bool XclExpXmlStream::exportDocument() throw()
 {
     ScDocShell* pShell = getDocShell();
     ScDocument* pDoc = pShell->GetDocument();
-    SotStorageRef rStorage = dynamic_cast <SotStorage*>( Reference<XStorage>( pShell->GetStorage() ).get() );
+    SfxMedium* pMedium = pShell->GetMedium();
+    SvStream* pMediumStream = pMedium->GetOutStream();
+    SotStorageRef rStorage = new SotStorage( pMediumStream, false );
 
     XclExpRootData aData( EXC_BIFF8, *pShell->GetMedium (), rStorage, *pDoc, RTL_TEXTENCODING_DONTKNOW );
     aData.meOutput = EXC_OUTPUT_XML_2007;
-- 
1.6.0.rc1.49.g98a8

